### PR TITLE
Updating instructions and docker-compose for running on Docker

### PR DESCRIPTION
**Pull request by santanaluiz**
#### Board:
* #599
---
#### Description:

Updating instructions for running on Docker
---
#### Notes:
* Updated README for clarity
* Added env variable to docker-compose file
* Added env variable to .env.sample file

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -13,6 +13,7 @@ services:
       POSTGRES_USER: ${POSTGRES_USER-postgres}
       POSTGRES_PASSWORD: ${POSTGRES_PASSWORD-postgres}
       POSTGRES_HOST: db
+      SECRET_KEY_BASE: ${SECRET_KEY_BASE-1234567890}
```
Comment (JulianPasquale) [inline docker-compose.yml:16]: Hey @santanaluiz, just curious here, are you running the app in production mode using `docker-compose`?

Our compose file is meant to be used for development purposes only, and I don't think the secret_key_base is needed in those cases.

Comment (santanaluiz) [reply docker-compose.yml:16]: I am not using for production, only dev env.
Running rspec fails to me if I don't add it

Comment (JulianPasquale) [reply docker-compose.yml:16]: This doesn't sound correct. According to the [Rails docs](https://edgeguides.rubyonrails.org/security.html#session-storage):

> In test and development applications get a secret_key_base derived from the app name

I don't think this is needed to run the tests.

Comment (JulianPasquale) [reply docker-compose.yml:16]: This is weird, I cannot reproduce it. I'm actually getting an error but doesn't seem to be related to the secret key

Comment (JulianPasquale) [reply docker-compose.yml:16]: @santanaluiz can you please check if this PR fixes your tests https://github.com/rootstrap/rails_api_base/pull/603?

**Comment by santib** [comment]
We just upgraded to Ruby 3.3.1 so the notice is no longer needed. Regarding SECRET_KEY_BASE, seems like it can't be reproduced, so closing the PR for now.

**Pull request closed without merging**